packages/core-data/src/private-selectors.ts

```diff
@@ -105,3 +106,129 @@ export function getEntityRecordPermissions(
 export function getRegisteredPostMeta( state: State, postType: string ) {
 	return state.registeredPostMeta?.[ postType ] ?? {};
 }
+
+interface SiteData {
+	show_on_front?: string;
+	page_on_front?: string | number;
+	page_for_posts?: string | number;
+}
+
+export const getHomePage = createRegistrySelector( ( select ) => () => {
+	const siteData = select( STORE_NAME ).getEntityRecord( 'root', 'site' ) as
+		| SiteData
+		| undefined;
+	const homepageId =
+		siteData?.show_on_front === 'page' &&
+		[ 'number', 'string' ].includes( typeof siteData.page_on_front ) &&
+		siteData.page_on_front &&
+		!! +siteData.page_on_front // We also need to check if it's not zero(`0`).
+			? siteData.page_on_front.toString()
+			: null;
+	const isSiteLoaded = !! siteData;
+	if ( isSiteLoaded && homepageId ) {
+		return { postType: 'page', postId: homepageId };
+	}
+	if ( isSiteLoaded && ! homepageId ) {
+		const frontPageTemplateId = select( STORE_NAME ).getDefaultTemplateId( {
+			slug: 'front-page',
+		} );
+		return { postType: 'wp_template', postId: frontPageTemplateId };
+	}
+	return null;
+} );
+
+export const getPostsPageId = createRegistrySelector( ( select ) => () => {
+	const siteData = select( STORE_NAME ).getEntityRecord( 'root', 'site' ) as
+		| SiteData
+		| undefined;
+	return siteData?.show_on_front === 'page' &&
+		siteData.page_for_posts &&
+		[ 'number', 'string' ].includes( typeof siteData.page_for_posts )
+		? siteData.page_for_posts.toString()
+		: null;
+} );
+
+export const getTemplateId = createRegistrySelector(
+	( select ) => ( state, postType, postId ) => {
+		const homepage = unlock( select( STORE_NAME ) ).getHomePage();
+
+		// For the front page, we always use the front page template if existing.
+		if (
+			postType === 'page' &&
+			postType === homepage?.postType &&
+			postId === homepage?.postId
+		) {
+			// The /lookup endpoint cannot currently handle a lookup
+			// when a page is set as the front page, so specifically in
+			// that case, we want to check if there is a front page
+			// template, and instead of falling back to the home
+			// template, we want to fall back to the page template.
+			const templates = select( STORE_NAME ).getEntityRecords(
+				'postType',
+				'wp_template',
+				{
+					per_page: -1,
+				}
+			);
+			if ( templates ) {
```

Comment:
This can also have better code style: `if ( ! templates ) return undefined` and then everything that follows can have less indentation and fewer `?.` operators.